src/Features/TerrainHelper.cpp

```diff
@@ -0,0 +1,143 @@
+#include "TerrainHelper.h"
+
+#include "ShaderCache.h"
+#include "State.h"
+
+void TerrainHelper::DataLoaded()
+{
+	// Get the default landscape texture set for terrain helper
+	const auto defaultLandTextureSet = RE::TESForm::LookupByEditorID<RE::BGSTextureSet>("LandscapeDefault");
+	if (defaultLandTextureSet != nullptr) {
+		spdlog::info("[Terrain Helper] LandscapeDefault EDID texture set found");
```

Comment:
Please propagate changes like the logger and nullptr checks throughout. I won't be checking all lines again.